scaper/core.py

```diff
@@ -1320,6 +1320,8 @@ def _instantiate_event(self, event, isbackground=False,
         # source file's duration without modification.
         if label in self.protected_labels:
             event_duration = source_duration
+        elif isbackground:
+            event_duration = self.duration
         else:
```

Comment:
Pedantic I know, but let's reorder this to:

if isbackground:
...
elif label in self.protected_albels:
...
else:
...